nav2_route/CMakeLists.txt

```diff
@@ -111,6 +111,14 @@ ament_target_dependencies(graph_file_loaders
 
 pluginlib_export_plugin_description_file(nav2_route plugins.xml)
 
+# Breadth first search library
+add_library(breadth_first_search SHARED
```

Comment:
Reason this is a separate library? I'm actually not sure what impact this has on compute time - I'd think potentially faster. I tend to just stick everything into a single library that doesn't need to be explicitly separated like plugins. That might not be good, but I haven't experimented with it.

Reply:
I have no idea how it affects comp time. I had a seperate lib for the test. I didn't think there was any need to link against other files that I don't need for the test.